src/components/calcite-alert/calcite-alert.tsx

```diff
@@ -79,6 +80,14 @@ export class CalciteAlert {
 
   connectedCallback(): void {
     if (this.active && !this.queued) this.calciteAlertRegister.emit();
+    // if an icon string is not provided, but icon is true and a default icon is present
+    // for the requested color, set that as the icon
+    const colors = ["blue", "green", "red", "yellow"];
+    this.icon = this.icon
```

Comment:
What if you have a stateful private property called `computedIcon` that watches `icon` and handles the boolean case?

Currently, if icon changes after initialization, the boolean case isn't being handled.

Thoughts?

Reply:
Yes, good catch. Will update on these new additions and the input component as well.

Reply:
Maybe move colors const to the resources file too. Could be used for testing

Reply:
Would it not be fine to just do this in `renderIcon`?  Seems pretty lightweight to handle this logic during render...

Reply:
Yeah that's another option. Either way works since its lightweight

Reply:
Agree that since alert and notice use the same logic it's a good idea to share in a resources file with some quick spec tests, that would be awesome

Reply:
Sounds good, I'll do some clean up on this.